[e2e tests] Check email for completed order

alvarothomas opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes [/woocommerce/woocommerce-quality/issues/656](https://github.com/woocommerce/woocommerce-quality/issues/656)

This updates order-emails with a test to verify that a completed order email was received and its content. The plug-in WP Mail Logging is used along these tests too.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run `pnpm test:e2e-pw ./tests/e2e-pw/tests/merchant/order-emails.spec.js --headed` and verify that tests pass.
2. Code review and verify approach.
3. Make sure that CI jobs are ✅

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message

This aims to add more checks to confirm that a completed order email is received, as well as its content.

#### Comment

</details>

Commit:
Add completed order email test

Add test structure to verify completed order email arrival and content

Commit:
Verify Email JSON Content

Parse email content in JSON format

Commit:
Record email content in JSON

Record email content in JSON

Commit:
Update emails completed order checks

Added checks to verify the content of completed order's emails

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

lanej0 commented:
@alvarothomas -- It looks like you've got a conflict on this branch.  Mind resolving that, and then I'll give this a quick review/merge?

Commit:
Merge branch 'trunk' into Check-email-for-completed-order

veljkho commented on plugins/woocommerce/tests/e2e-pw/tests/merchant/order-emails.spec.js:
```diff
@@ -97,6 +97,68 @@ test.describe( 'Merchant > Order Action emails received', () => {
+			page.locator( '#wp-mail-logging-modal-content-body-content' )
+		).toContainText( 'Thanks for shopping with us' );
+
+		// Verify that the email contains a "Thanks" note
+		await expect(
+			page.locator( '#wp-mail-logging-modal-content-body-content' )
+		).toContainText( 'We have finished processing your order.' );
```
This is a duplicate assertion; you already have one at the 137-139 line.

alvarothomas replied:
Oh, yeah! I think this got added when I solved the conflict in the branch. Thanks for the heads-up.

veljkho commented on plugins/woocommerce/tests/e2e-pw/tests/merchant/order-emails.spec.js:
```diff
@@ -97,6 +97,68 @@ test.describe( 'Merchant > Order Action emails received', () => {
+
+		// Verify that the email contains the order ID
+		await expect(
+			page.locator( '#wp-mail-logging-modal-content-body-content' )
```
You could const the selector at the top of this test since it has been used several times:
`const emailBodyContent = '#wp-mail-logging-modal-content-body-content';`

alvarothomas replied:
Good idea!

veljkho reviewed (changes requested):
I left some comments @alvarothomas 👍

Commit:
Update email content const & remove duplicated test

Updated with a new const to record email content const and removed duplicated assertion.

alvarothomas commented:
I've committed the changes you suggested @veljkho , thank you so much for the review 👍🏻

veljkho reviewed (approved):
LGTM
